discord_connector.py

```diff
@@ -0,0 +1,346 @@
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_response(r, action_result)
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+            r.status_code,
+            r.text.replace('{', '{{').replace('}', '}}')
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        config = self.get_config()
+
+        resp_json = None
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)),
+                resp_json
+            )
+
+        # Create a URL to connect to
+        url = self._base_url + endpoint
+
+        try:
+            r = request_func(
+                url,
+                # auth=(username, password),  # basic authentication
+                verify=config.get('verify_server_cert', False),
+                **kwargs
+            )
+        except Exception as e:
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(str(e))
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self._loop.run_until_complete(self._client.login(self._token))
+
+        if self._client.status != discord.Status.online:
+            self.save_progress("Test Connectivity Failed.")
+            return action_result.get_status()
+
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    async def _handle_get_guild(self):
+        await self._client.login(self._token)
+        self._guild = await self._client.fetch_guild(self._guild_id)
+
+    def _handle_list_channels(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress(str(type(self._guild)))
+
+        channels = self._loop.run_until_complete(self._guild.fetch_channels())
+        num_channels = 0
+
+        for channel in channels:
+            if type(channel) == discord.TextChannel:
+                num_channels += 1
+                add_chan = {}
+                add_chan['name'] = channel.name
+                add_chan['id'] = channel.id
+                action_result.add_data(add_chan)
```

Comment:
Can be simplified into:
```suggestion
                action_result.add_data({
                    "name": channel.name,
                    "id": channel.id
                })
```